pkg/controller/gitlab/kubernetes_test.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2019 The GitLab-Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gitlab
+
+import (
+	"context"
+	"testing"
+
+	xpcomputev1alpha1 "github.com/crossplaneio/crossplane/pkg/apis/compute/v1alpha1"
+	xpcorev1alpha1 "github.com/crossplaneio/crossplane/pkg/apis/core/v1alpha1"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplaneio/gitlab-controller/pkg/apis/controller/v1alpha1"
+	"github.com/crossplaneio/gitlab-controller/pkg/test"
+	"github.com/crossplaneio/gitlab-controller/pkg/util"
+)
+
+func Test_kubernetesReconciler_reconcile(t *testing.T) {
+	ctx := context.TODO()
+	testError := errors.New("test-error")
+	type fields struct {
+		base   *baseResourceReconciler
+		finder resourceClassFinder
+	}
+	type want struct {
+		err    error
+		status *xpcorev1alpha1.ResourceClaimStatus
+	}
+	tests := map[string]struct {
+		fields fields
+		want   want
+	}{
+		"FailToFindResourceClass": {
+			fields: fields{
+				base: &baseResourceReconciler{
+					GitLab: newGitLabBuilder().build(),
+				},
+				finder: &mockResourceClassFinder{
+					mockFind: func(ctx context.Context, provider v1.ObjectReference, resource string) (*v1.ObjectReference, error) {
+						return nil, testError
+					},
+				},
+			},
+			want: want{err: errors.Wrapf(testError, errorFmtFailedToFindResourceClass, kubernetesClaimKind,
+				newGitLabBuilder().build().GetProviderRef())},
+		},
+		"FailToCreate": {
+			fields: fields{
+				base: &baseResourceReconciler{
+					GitLab: newGitLabBuilder().withMeta(testMeta).build(),
+					client: &test.MockClient{
+						MockGet: func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+							return kerrors.NewNotFound(schema.GroupResource{}, "")
+						},
+						MockCreate: func(ctx context.Context, obj runtime.Object) error {
+							return testError
+						},
+					},
+				},
+				finder: &mockResourceClassFinder{
+					mockFind: func(ctx context.Context, provider v1.ObjectReference, resource string) (*v1.ObjectReference, error) {
+						return nil, nil
+					},
+				},
+			},
+			want: want{err: errors.Wrapf(testError, errorFmtFailedToCreate, kubernetesClaimKind, testKey)},
+		},
+		"FailToRetrievePostgresObject-Other": {
```

Comment:
this file is testing kubernetes resources, is the postgres reference here a copy/paste issue? i see it in some of the other test files too

Reply:
ah.. yes, must be. checking...